consensus/hotstuff/eventhandler/event_handler.go

```diff
@@ -334,11 +334,15 @@ func (e *EventHandler) proposeForNewViewIfPrimary() error {
 	if e.committee.Self() != currentLeader {
 		return nil
 	}
-	for _, b := range e.forks.GetBlocksForView(curView) {
+	for _, b := range e.forks.GetBlocksForView(curView) { // on the happy path, this slice is empty
 		if b.ProposerID == e.committee.Self() {
 			log.Debug().Msg("already proposed for current view")
 			return nil
 		}
+		// sanity check: the following code should never be reached, as this node is the current leader, i.e.
+		// we should _not_ consider a proposal for this view from any other as valid and store it in forks.
+		//nolint:staticcheck
+		return fmt.Errorf("this node (%v) is leader for the current view %d, but have a proposal from node %v for this view", currentLeader, curView, b.ProposerID)
```

Comment:
```suggestion
		} else {
		// sanity check: the following code should never be reached, as this node is the current leader, i.e.
		// we should _not_ consider a proposal for this view from any other as valid and store it in forks.
		return fmt.Errorf("this node (%v) is leader for the current view %d, but have a proposal from node %v for this view", currentLeader, curView, b.ProposerID)
		}
```

Reply:
adding an else statement removes need for `nolint`

Reply:
✅ 👉 https://github.com/onflow/flow-go/pull/4392/commits/0ea87bf5e62d170ab520e970943c5d551f0df204